### PR TITLE
ci: fix evaluating pr data and review status + minor tweaks

### PR DESCRIPTION
**Pull request by nashif**
- ci: merged_pr stats: fix a bug an minor tweaks
- ci: pr stats: define elasticsearch index as a repo variable

### CODE REVIEW

**Review by stephanosio — DISMISSED** [review]

File: scripts/ci/stats/merged_prs.py [hunk 1]
```diff
@@ -84,11 +74,28 @@ def main():
...
 
-        ll = []
-        for l in pr.labels:
-            ll.append(l.name)
-        prj['labels'] = ll
+        for assignee in pr.assignees:
+            if assignee.login in reviewers:
+                assignee_reviews += 1
+                # was reviewed at least by one assignee
+                prj['reviewed_by_assignee'] = "yes"
```
Comment (kartben) [inline scripts/ci/stats/merged_prs.py:96]: it's never set to "no" anywhere? maybe also initialize with a default value alongside others? I guess default is "na"?

Comment (kartben) [reply scripts/ci/stats/merged_prs.py:96]: Also just thinking out loud know, but I realize that Kibana (or any other reporting tool we'd want to hook up to the raw data, really) is probably smart enough to allow us to write the rules there, so wouldn't it be better to just make sure we capture *raw* data in ES (ex. "# of assignees", "# of approving assignees", etc), and add computed columns/fields in Kibana?

I guess having the rules visible here in the script is nice as it's easier for everyone to see and now how the dashbboards are then put together, but on the other it feels a bit like we might be making things more complex to update than necessary.

File: scripts/ci/stats/merged_prs.py [hunk 2]
```diff
@@ -98,42 +105,44 @@ def main():
...
-        # go through all assignees and check if anyone has approved and reset assignee to the one who approved
+        # go through all assignees and check if assignee is the submitter
         for assignee in pr.assignees:
-            if assignee.login in reviewers:
-                prj['assignee'] = assignee.login
-            elif assignee.login == pr.user.login:
-                prj['reviewed_by_assignee'] = "yes"
-
+            if  assignee.login == pr.user.login:
+                prj['review_rule'] = "yes"
```
Comment (kartben) [inline scripts/ci/stats/merged_prs.py:111]: sorry, what's this rule supposed to be, I'm drawing a blank here? And in any case can't the check be performed as part of the earlier `for assignee in pr.assignees` loop?

File: scripts/ci/stats/merged_prs.py [hunk 3]
```diff
@@ -84,11 +74,28 @@ def main():
...
+        for assignee in pr.assignees:
+            if assignee.login in reviewers:
+                assignee_reviews += 1
+                # was reviewed at least by one assignee
+                prj['reviewed_by_assignee'] = "yes"
+                prj['assignee'] = assignee.login
+                prj['review_rule'] = "yes"
 
         # take first assignee, otherwise we have no assignees and this rule is not applicable
         if pr.assignee:
```
Comment (kartben) [inline scripts/ci/stats/merged_prs.py:101]: could you try and have all check performed against `pr.assignees` and not a mix of `pr.assignee` / `pr.assignees`? This would probably allow to group a few if blocks together too?

**Review by kartben — CHANGES_REQUESTED** [review]
Thanks for adding comments and making some variable names more explicit!
afaict there is probably a bug since `reviewed_by_assignee` is never set to no?
A few other comments/questions as well. Thanks!

File: scripts/ci/stats/merged_prs.py (hunk 1 above)
Comment (nashif) [reply scripts/ci/stats/merged_prs.py]: > it's never set to "no" anywhere? maybe also initialize with a default value alongside others? I guess default is "na"?

removed completely, we have this already covered by `assignee_reviews` count.

File: scripts/ci/stats/merged_prs.py (hunk 2 above)
Comment (nashif) [reply scripts/ci/stats/merged_prs.py]: addressed

File: scripts/ci/stats/merged_prs.py (hunk 3 above)
Comment (nashif) [reply scripts/ci/stats/merged_prs.py]: addressed

File: scripts/ci/stats/merged_prs.py [hunk 4]
```diff
@@ -84,56 +74,72 @@ def main():
...
+                assignee_reviews += 1
+                # was reviewed at least by one assignee
                 prj['assignee'] = assignee.login
-            elif assignee.login == pr.user.login:
-                prj['reviewed_by_assignee'] = "yes"
+                prj['review_rule'] = "yes"
+
+            # if assignee is the submitter, rule does not apply.
+            if  assignee.login == pr.user.login:
+                prj['review_rule'] = "na"
```
Comment (fabiobaltieri) [inline scripts/ci/stats/merged_prs.py:101]: I think this could result in "na" or "yes" depending on the order of assignee, maybe just set `review_rule` once outside of the loop depending on the value of `assignee_reviews`? It's also hard to follow right now sicne this can be overridden by a bunch of different code paths

Comment (nashif) [reply scripts/ci/stats/merged_prs.py]: updated

**Review by fabiobaltieri — APPROVED** [review]

**Review by kartben — APPROVED** [review]
